### ISSUE TITLE
Side effet of the plugin - follow up

### ISSUE THREAD

**Issue by MOEDevOps** [mesilat/confluence-fields#20]
Hi,
Following our issue on #18 we still have an issue with Jira service desk. The page does not load correctly initially. Users must force the refresh of the page to get it displayed correctly.

![Capture-1](https://user-images.githubusercontent.com/64581011/82199532-18002800-98fe-11ea-836a-97507bacfd2c.jpg)

**Comment by mesilat** [comment]
Hi, thanks for reporting.
Could you please also attach browser logs?

**Comment by MOEDevOps** [comment]
This is the log for the screenshot displayed in this comment [jira.txt](https://github.com/mesilat/confluence-fields/files/4649554/jira.txt):

![screenshot](https://user-images.githubusercontent.com/64581011/82311477-0da96100-99c6-11ea-8f89-3de0fb6fe88e.png)

**Comment by mesilat** [comment]
Hi,
Unfortunately I could not figure out what goes wrong: there are 7 web requests logged and all of them succeeded (status 200), and none is using the plugin endpoints. Thus the problem should be somewhere in javascript but you provided network log, not console log.

At the same time the screenshot implies there are 11 warnings in the log and no errors. This is strange as there should be some script error that causes jira to fail. Could you please provide the console log as well? I need to see an error :)

**Comment by MOEDevOps** [comment]
Hello, 

My apologizes, I'll try to be more precise.
1) Going into JIRA SD and clicking on the link (i.e. "Question ou Demande") hangs up the application (screenshot-1.png)
2) After refreshing (by pressing F5) the page, the application continues to work "correctly" (screenshot-2.png and logfile-2.txt)
3) I disable the plugin on JIRA Apps manager.
4) Goint back to JIRA SD and clicking again on the link (i.e. "Question ou Demande") the application displays the page correctly  (screenshot-3.png and logfile-3.txt)

Hope this could help you to see where is the issue ?

![screenshot-1](https://user-images.githubusercontent.com/64581011/82409862-f83c4180-9a6e-11ea-9608-a1fca4738e69.png)
![screenshot-2](https://user-images.githubusercontent.com/64581011/82409883-00947c80-9a6f-11ea-9d3d-dc3e7a78517a.png)
[logfile-2.txt](https://github.com/mesilat/confluence-fields/files/4654866/logfile-2.txt)
![screenshot-3](https://user-images.githubusercontent.com/64581011/82409974-30dc1b00-9a6f-11ea-818a-16a9d5a6129e.png)
[logfile-3.txt](https://github.com/mesilat/confluence-fields/files/4654870/logfile-3.txt)

**Comment by mesilat** [comment]
Sorry, but there's nothing that could give the clue. I will install your JSD version and try to reproduce the issue.
Meanwhile, could you please confirm that this is anonymous portal? Just a guess...

**Comment by MOEDevOps** [comment]
Hi, 
Apparently this issue happens for any user accessing directly into the JIRA SD portal. We try with users role "Service Desk Customers" or user having a JSD licence, both produce the same effect.

The difference between the two log files shows deprecated JavaScript functions (i.e. chrome.loadTimes()) ... is it an indication ?

King Regards
Philippe Martin

**Comment by mesilat** [comment]
Hi, Philippe
Deprecated APIs are logged by some other SD components and should not be a problem.

Please try version 1.0.28 from this repo (https://github.com/mesilat/confluence-fields/releases/tag/1.0.28) -- I did some refactoring to portal code, hope this will resolve the issue.

If not, then please open your browser console and enable the plugin trace logging by typing `localStorage.CFP_TRACE_ENABLED = true`
Then refresh the page and reproduce the issue. Paste here all the log from the console.

Best regards,
Vladislav Babin

**Comment by MOEDevOps** [comment]
Hi Vladislav, 

Following your suggestion, I insist to produce an error. There is one. Following the above process, I press on the back button (instead of pressing F5) ... an error can be seen in to screenshot-4.png image and in the logfile-4.txt. 

N.B. This eror does not appears when the plugin is disabled.

Does this help ?

King regards
Philippe

![screenshot-4](https://user-images.githubusercontent.com/64581011/83026206-809e8180-a02f-11ea-8b63-1736a26c3455.png)
[logfile-4.txt](https://github.com/mesilat/confluence-fields/files/4689052/logfile-4.txt)

**Comment by MOEDevOps** [comment]
One more thing, I install the version 1.0.28 but this does not change the problem.

**Comment by mesilat** [comment]
Not really, as I do not use xsrfToken anywhere in my code. Screenshot shows normal confluence-fields flow then it breaks before loading the plugin settings from JIRA server.
I introduced some more trace log in portal.js module, so I will know exactly the line that fails. Could you please install version 1.0.29 and attach the logs? Thank you
Best regards,
Vladislav Babin

**Comment by MOEDevOps** [comment]
Hi Vladislav, 

Here the log files : 
1) produced after going into JIRA SD et click on the link  "Question ou Demande"; hangs up
[jira.vdad1.bcv.ch-1.log](https://github.com/mesilat/confluence-fields/files/4699996/jira.vdad1.bcv.ch-1.log)
2) produced after refreshing the browser by typing F5
[jira.vdad1.bcv.ch-2.log](https://github.com/mesilat/confluence-fields/files/4699998/jira.vdad1.bcv.ch-2.log)
3) produced after click onback button; hangs up
[jira.vdad1.bcv.ch-3.log](https://github.com/mesilat/confluence-fields/files/4699999/jira.vdad1.bcv.ch-3.log)

**Comment by mesilat** [comment]
The logs look normal I can see nothing unexpected and indication of an error, sorry.

However I have started to observe the similar behaviour to what you describe in my demo setup :)

From what I could see so far the problem is in Atlassian WRMPlugin code loading, which does not play well with customer portal. I would have to either refactor my portal.js module to not use WRMPlugin or find a way to make it behave. Either is not an easy choice.

Meanwhile you could disable the plugin's portal module to get back your customer portal:

<img width="1064" alt="Screenshot 2020-06-02 at 08 15 12" src="https://user-images.githubusercontent.com/18350871/83482639-6ccca280-a4a9-11ea-824e-b21caeeaf0eb.png">

**Comment by MOEDevOps** [comment]
Hi, 
Great ! I think we have something... Disabling the plugin **value-converter-portal-rest-filter** (not entrypoint-confluence-fields-portal) does the job ! :-)

Since we don't use this part of the plugin this means this issue has a lower priority (for us).

Kind regards
Philippe

![screenshot-5](https://user-images.githubusercontent.com/64581011/83495735-6560c580-a4b8-11ea-9736-e426eb819ca2.png)
